trackintel/analysis/tracking_quality.py

```diff
@@ -106,13 +109,19 @@ def temporal_tracking_quality(source, granularity="all", max_iter=60):
         # get the tracked week relative to the first day
         start_date = df["started_at"].min().date()
         df["week"] = df["started_at"].apply(lambda x: (x.date() - start_date).days // 7)
-        dict_week_num = dict(zip(df["week"], df["started_at"].apply(lambda x: pd.to_datetime(x).week)))
+        dict_start_week_num = dict(
+            zip(df["week"], df["started_at"].apply(lambda x: pd.to_datetime(x).isocalendar()[1]))
+        )
+        dict_finish_week_num = dict(
+            zip(df["week"], df["finished_at"].apply(lambda x: pd.to_datetime(x).isocalendar()[1]))
+        )
+        dict_week_num = {**dict_start_week_num, **dict_finish_week_num}
 
         # calculate per-user per-week raw tracking quality
         raw_quality = df.groupby(["user_id", "week"], as_index=False).apply(_get_tracking_quality_user, granularity)
         # add quality = 0 records
         quality = _get_all_quality(df, raw_quality, granularity)
-        quality["week_number"] = quality["week"].map(dict_week_num)
+        quality["week_number"] = quality["week"].map(dict_week_num).astype("int32")
```

Comment:
Last time we agree that we still return a DateTime format value when `granularity == "week"`? This means this should be something like the DateTime object of the first day of the week, which ensures flexibility. 

This would also mean we need to change the  `"week_number"` into something like `week_monday`

Reply:
done, the doctoring updated accordingly

Reply:
I think this still does not work on my PC... the returned date is in format datatime.date, and not datatime.datetime (the pandas datatype for time). The returned column is therefore not usable using .dt method. I do not know how to change this, but I find using pd.Grouper pandas handled this for us